components/notebook-controller/pkg/controller/notebook/notebook_controller.go

```diff
@@ -216,6 +219,28 @@ func (r *ReconcileNotebook) Reconcile(request reconcile.Request) (reconcile.Resu
 		}
 	}
 
+	// Reconcile virtual service
+	virtualService, err := generateVirtualService(instance)
+	if err := controllerutil.SetControllerReference(instance, virtualService, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+	// Check if the virtual service already exists. Noop if it already exists.
+	foundVirtual := &unstructured.Unstructured{}
+	foundVirtual.SetAPIVersion("networking.istio.io/v1alpha3")
+	foundVirtual.SetKind("VirtualService")
+	err = r.Get(context.TODO(), types.NamespacedName{Name: virtualServiceName(instance.Name,
+		instance.Namespace), Namespace: instance.Namespace}, foundVirtual)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Creating virtual service", "namespace", instance.Namespace, "name",
+			virtualServiceName(instance.Name, instance.Namespace))
+		err = r.Create(context.TODO(), virtualService)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
you need an update step here, see 213~220 for service, or 185~193 for ss

Reply:
doesn't seem to have a proper way of doing it - having this error when trying to run update:
`"error":"virtualservices.networking.istio.io \"notebook-kubeflow-nb-testing\" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update`

Reply:
added.